static/js/public-adventures.ts

```diff
@@ -101,5 +101,5 @@ document.addEventListener("updateTSCode", (e: any) => {
         initialize({lang: js.lang, level: parseInt(js.level), keyword_language: js.lang,
             javascriptPageOptions: js
             });
-    }, 1000);
+    }, 500);
```

Comment:
Rendering takes some time and if i initialize the js, some parts wouldn't work. The 1000 is just a longer, unneeded duration

Reply:
Yeah, but fighting race conditions with sleeps isn't a great idea.

Imagine this website running on a slightly slower computer -- the init functions is going to run too quickly again, before all the initialization that you depend on has happened. You're never going to cover all computer speeds with one timeout.

It would be better to find a more deterministic point to do this:

- (Preferred) find some way to hook into the JavaScript that you need to run after, and make it call you when it's done. 
- (Not preferred but possible) Poll, as much as it sucks! Set an interval timer to run every 50ms, do some check against the DOM to see if an HTML element you depend on exists, and if it does, and do your initialization. Don't forget to cancel the timer! 

Now your initialization is going run ~more or less as quickly as it can.

Reply:
> * (Preferred) find some way to hook into the JavaScript that you need to run after, and make it call you when it's done.

What could work is to listen on an event that's fired after the dom elements are replaced and then update the js. Will try that!